controllers/controlplane.go

```diff
@@ -86,7 +95,7 @@ func (c *ControlPlane) MachinesNeedingRollout() collections.Machines {
 func getInfraResources(ctx context.Context, cl client.Client, machines collections.Machines) (map[string]*unstructured.Unstructured, error) {
 	result := map[string]*unstructured.Unstructured{}
 	for _, m := range machines {
-		infraObj, err := external.Get(ctx, cl, &m.Spec.InfrastructureRef, m.Namespace)
+		infraObj, err := external.Get(ctx, cl, &m.Spec.InfrastructureRef, m.Spec.InfrastructureRef.Namespace)
```

Comment:
Is this an intentional change? 🤔 
If `m.Spec.InfrastructureRef.Namespace` is not specified (it's optional), it will result in an empty string

Reply:
You are right. Removed it.